src/variableProcessor.ts

```diff
@@ -198,16 +198,62 @@ export class VariableProcessor {
+                                     * name (e.g. "Foo" in "Foo Bar"). If the search fails, the first domain is used.
+                                     */
+                                    let displayName = element.displayName;
                                     const count = element.domains.length;
-                                    const domain = `${element.domains[0]}${count > 1 ? " (+" + (count - 1) + " more)" : ""}`;
+                                    let domain = element.domains && element.domains[0];
+                                    if (count > 1) {
+                                        try {
+                                            // find the best matches
+                                            const displayNameSpaceIndex = displayName.indexOf(" ");
+                                            const displayNameFirstWord = displayNameSpaceIndex > -1
+                                                ? displayName.substring(0, displayNameSpaceIndex)
+                                                : displayName;
+                                            let bestMatches = [];
+                                            for (let d in element.domains) {
+                                                const matches = new RegExp(`(^${displayNameFirstWord}\.com$)|(^${displayNameFirstWord}\.[a-z]+(?:\.[a-z]+)?$)|(^${displayNameFirstWord}[a-z]+\.com$)|(^${displayNameFirstWord}[^:]*$)|(^[^:]*${displayNameFirstWord}[^:]*$)`, "gi")
+                                                    .exec(element.domains[d]) || [null, null, null, null, null, null];
+
+                                                // stop looking if the best match is found
+                                                bestMatches[0] = bestMatches[0] || matches[1];
+                                                if (bestMatches[0]) {
+                                                    break;
+                                                }
+
+                                                bestMatches[1] = bestMatches[1] || matches[2];
+                                                bestMatches[2] = bestMatches[2] || matches[3];
+                                                bestMatches[3] = bestMatches[3] || matches[4];
+                                                bestMatches[4] = bestMatches[4] || matches[5];
+                                            }
+
+                                            // use best match
+                                            for (let m in bestMatches) {
+                                                if (bestMatches[m]) {
+                                                    domain = bestMatches[m];
+                                                    break;
+                                                }
+                                            }
+                                        } catch {

```

Comment:
Can you run successfully locally? I think JS needs an expression for catch statements, refer to https://docs.microsoft.com/en-us/scripting/javascript/reference/try-dot-dot-dot-catch-dot-dot-dot-finally-statement-javascript

Reply:
Yep, everything works fine. I didn't see anything in the docs that indicted we need something else. Did I miss it? I'll add `catch (e)` just in case.

Reply:
I think JS needs us to write like `catch (e)` instead of `catch`?